File: ios/EventEmitter.m
//
//  EventEmitter.m
//  BlueWallet
//
//  Created by Marcos Rodriguez on 12/25/20.
//  Copyright © 2020 BlueWallet. All rights reserved.
//

#import "EventEmitter.h"

static EventEmitter *sharedInstance;

@implementation EventEmitter

RCT_EXPORT_MODULE();

+ (BOOL)requiresMainQueueSetup {
    return YES;
}

+ (instancetype)sharedInstance {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (void)removeListeners:(double)count {
  
}

- (instancetype)init {
  self = [super init];
  return self;
}

- (NSArray<NSString *> *)supportedEvents {
    return @[@"onUserActivityOpen"];
}

- (void)sendUserActivity:(NSDictionary *)userInfo
{
  [self sendEventWithName:@"onUserActivityOpen" body:userInfo];
}

RCT_EXPORT_METHOD(getMostRecentUserActivity:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
{
    NSUserDefaults *defaults = [[NSUserDefaults alloc] initWithSuiteName:@"group.io.bluewallet.bluewallet"];
    resolve([defaults valueForKey:@"onUserActivityOpen"]);
}

@end

File: ios/EventEmitter.swift
import Foundation
import React

@objc(EventEmitter)
class EventEmitter: RCTEventEmitter {
    static let sharedInstance = EventEmitter()
    
    override class func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc static func shared() -> EventEmitter {
        return sharedInstance
    }
    
    override func supportedEvents() -> [String]! {
        return ["onUserActivityOpen"]
    }
    
    @objc func sendUserActivity(_ userInfo: [String: Any]) {
        // Removed unnecessary type check; directly sending the event with userInfo.
        sendEvent(withName: "onUserActivityOpen", body: userInfo)
    }
    
    @objc func getMostRecentUserActivity(_ resolve: @escaping RCTPromiseResolveBlock,
                                           rejecter reject: RCTPromiseRejectBlock) {
        if let defaults = UserDefaults(suiteName: "group.io.bluewallet.bluewallet") {
            resolve(defaults.value(forKey: "onUserActivityOpen"))
        } else {
            resolve(nil)
        }
    }
}
